service/tx/service.go

```diff
@@ -65,16 +67,39 @@ func NewService(
 	}
 }
 
+func getSendTxContext(args client.SendTxRequest) (data []byte, t action.Type, err error) {
+	if len(args.Data) != 0 {
+		// means that we execute a contract method
+		msg := action_sc.Execute{
+			From:   keys.Address(args.From),
+			Amount: args.Amount,
+			Data:   args.Data,
+			Nonce:  args.Nonce,
+		}
+		if len(args.To.Bytes()) != 0 {
+			to := keys.Address(args.To)
+			msg.To = &to
+		}
+		data, err = msg.Marshal()
+		t = action.SC_EXECUTE
+	} else {
+		// just a regular send
+		msg := transfer.Send{
+			From:   keys.Address(args.From),
+			To:     keys.Address(args.To),
+			Amount: args.Amount,
+		}
+		data, err = msg.Marshal()
+		t = action.SEND
+	}
+	return
+}
+
 // SendTx exists for maintaining backwards compatibility with existing olclient implementations. It returns
 // a signed transaction
 // TODO: deprecate this
 func (svc *Service) SendTx(args client.SendTxRequest, reply *client.CreateTxReply) error {
-	send := transfer.Send{
-		From:   keys.Address(args.From),
-		To:     keys.Address(args.To),
-		Amount: args.Amount,
-	}
-	data, err := send.Marshal()
+	data, t, err := getSendTxContext(args)
```

Comment:
the SendTx rpc was just simple send, maybe should make a separated rpc for contract?

Reply:
Made it similar to the ethereum logic.

Reply:
I also thought about to make it separatelly, but anyway it is more convenient to use in the sendTransaction.